### ISSUE TITLE
batch_norm didn't really distinguish training/testing

### ISSUE THREAD

**Issue by kaonashi-tyc** [carpedm20/DCGAN-tensorflow#107]
The current implementation of batch_norm seems problematic to me that it always assumes training mode.

**Comment by carpedm20** [comment]
Why do you think so? I think inference mode is defined like [this](https://github.com/carpedm20/DCGAN-tensorflow/blob/master/model.py#L423) and variable is shared with [this](https://github.com/carpedm20/DCGAN-tensorflow/blob/master/model.py#L320)

**Comment by kaonashi-tyc** [comment]
my mistake, close
